### PR TITLE
fix(await-async-utils): false positives

### LINKED ISSUE #118: await-async-utils false positives
The await-async-utils rule is finding function calls that aren't actually from testing library. For example, if I define a function `wait` in my test code:

```js
function wait(delay) {
  return new Promise((resolve) => {
    setTimeout(resolve, delay);
  });
}
```

The lint rule will match it since it has the same name as the testing library's async util.

Would it make sense to first detect the import of testing-library, before applying the rule?

### PR DESCRIPTION
**Pull request by dmitry-lobanov**
Closes #118; Related #122

Fixes false positives `no-await-sync-query`

### CODE REVIEW

File: lib/rules/await-async-utils.ts [hunk 1]
```diff
@@ -49,17 +49,49 @@ export default ESLintUtils.RuleCreator(getDocsUrl)<Options, MessageIds>({
   defaultOptions: [],
 
   create(context) {
-    const testingLibraryUtilUsage: TSESTree.Identifier[] = [];
+    const asyncUtilsUsage: Array<{ node: TSESTree.Identifier | TSESTree.MemberExpression, name: string }> = [];
+    const importedAsyncUtils: string[] = [];
+
     return {
+      'ImportDeclaration > ImportSpecifier,ImportNamespaceSpecifier'(node: TSESTree.Node) {
```
Comment (Belco90) [inline lib/rules/await-async-utils.ts:56]: More clear now ✨

File: tests/lib/rules/await-async-utils.test.ts [hunk 2]
```diff
@@ -114,33 +143,46 @@ ruleTester.run(RULE_NAME, rule, {
...
           doSomethingElse();
           ${asyncUtil}(() => getByLabelText('email'));
         });
       `,
-      errors: [{ line: 4, messageId: 'awaitAsyncUtil' }],
+      errors: [{ line: 5, messageId: 'awaitAsyncUtil' }],
+    })),
+    ...ASYNC_UTILS.map(asyncUtil => ({
+      code: `
+        import * as asyncUtil from '@testing-library/dom';
```
Comment (Belco90) [inline tests/lib/rules/await-async-utils.test.ts:156]: 💯

**Comment by dmitry-lobanov** [comment]
Hi @Belco90! Thanks, I'm glad to contribute.

**Pull request merged**